lib/rules/padding-between-test-functions.js

```diff
@@ -0,0 +1,142 @@
+
+'use strict';
+
+var names = [
+    'describe',
+    'context',
+    'it',
+    'specify',
+    'before',
+    'after',
+    'beforeEach',
+    'afterEach',
+    'suite',
+    'test',
+    'suiteSetup',
+    'suiteTeardown',
+    'setup',
+    'teardown'
+];
+
+/**
+ * Checks if a node is a test function expression, or object (e.g., it() or it.only())
```

Comment:
Good call, I didn't realize those utility functions existed.  Probably quite a bit cleaner than my `isTestFunction` routine. 

However, because it is split out into 3 different functions, I'm running into complexity problems again due to all the or branches...

I'm going to add the missing hooks, and another util, `isTestFunction` that returns true for any of them.